packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/SessionReplaySDKWrapper.kt

```diff
@@ -21,4 +24,10 @@ internal class SessionReplaySDKWrapper : SessionReplayWrapper {
             sessionReplayConfiguration,
         )
     }
+
+    internal companion object {
+        internal fun getLogger(): InternalLogger? {
+            return (Datadog.getInstance() as? FeatureSdkCore)?.internalLogger
```

Comment:
I'm not sure this is a best way to get a logger, because it has 2 problems:

* it introduces nullability (it can be avoided if we get logger only once when feature is initialized, because it is guaranteed that core will be there at this point and its type is `FeatureSdkCore`)
* each call to `Datadog.getInstance` introduces synchronization lock (not a big penalty, but still).

Normally in other features we are passing down the logger instance we got only once (or passing down SDK core instance), and anyway in the code of this module you have `InternalLogger` as constructor argument.

But it is not a blocker for me, if it works fine for you.

Reply:
Oh this one is on me, I suggested this change. 

I wasn't aware of the sync lock, let's watch how much of an impact this has in the perf benchmark.

Reply:
probably just peanuts and we shouldn't care about this penalty. But if we have some tracing data, it will be good.

Reply:
I've changed it back to passing the logger instance